browser/psst/psst_ui_delegate_impl.cc

```diff
@@ -5,16 +5,53 @@
 
 #include "brave/browser/psst/psst_ui_delegate_impl.h"
 
+#include "content/public/browser/web_contents.h"
+
 namespace psst {
 
-PsstUiDelegateImpl::PsstUiDelegateImpl() = default;
+PsstUiDelegateImpl::PsstUiDelegateImpl(
+    BravePsstPermissionContext* permission_context,
+    content::WebContents* contents)
+    : web_contents_(contents), permission_context_(permission_context) {
+  CHECK(permission_context);
+}
+
 PsstUiDelegateImpl::~PsstUiDelegateImpl() = default;
 
+void PsstUiDelegateImpl::Show(PsstConsentData dialog_data) {
+  dialog_data_ = std::move(dialog_data);
+
+  // Implementation for showing the consent dialog to the user.
+
+  // When dialog accepted by the user
+  OnUserAcceptedPsstSettings(base::Value::List());
+}
+
 void PsstUiDelegateImpl::UpdateTasks(
     long progress,
     const std::vector<PolicyTask>& applied_tasks,
     const mojom::PsstStatus status) {
   // Implementation for setting the current progress.
 }
 
+std::optional<PsstPermissionInfo> PsstUiDelegateImpl::GetPsstPermissionInfo(
+    const url::Origin& origin,
+    const std::string& user_id) {
+  return permission_context_->GetPsstPermissionInfo(origin, user_id);
+}
+
+void PsstUiDelegateImpl::OnUserAcceptedPsstSettings(
+    base::Value::List urls_to_skip) {
+  // Create the PSST permission when user accepts the dialog
+  permission_context_->GrantPermission(
+      url::Origin::Create(web_contents_->GetLastCommittedURL()),
```

Comment:
this is wrong. web contents can be navigated to another URL while the dialog is opened.

Reply:
this is part of my point here https://github.com/brave/brave-core/pull/31348#discussion_r2411819240 that there are a bunch of edge cases to handle for permission dialogs